packages/components/src/ListView/Items/Item/Item.component.js

```diff
@@ -23,23 +25,35 @@ function Item(props) {
 	}
 
 	const itemId = id && `checkbox-${id}`;
-	const itemSelector = isSwitchBox ? 'switch checkbox' : 'checkbox';
+	const itemClassName = classNames(
+		'checkbox',
+		{ 'switch-nested': props.children },
+		{ switch: isSwitchBox },
+	);
+
+	const onChange = parentItem
+		? event => item.onChange(event, item, parentItem)
+		: event => item.onChange(event, item);
```

Comment:
I'm not sure I understand the change you made.
What I meant was that you can call `item.onChange(event, item, parentItem)` on every cases. 
The item's `onChange` will handle it.